### PR TITLE
Feature/music

### PR DESCRIPTION
**Pull request by QuantumBits**
We now have a theme song! (The beginnings of one at least)

`File size of "dist/release.zip" is 7976/13312 bytes (59.91586538461539) %`

### CODE REVIEW

**Comment by QuantumBits** [comment]
Size reduced slightly! Even with more features!
`File size of "dist/release.zip" is 7957/13312 bytes (59.77313701923077) %`

**Comment by QuantumBits** [comment]
Got a tad bigger, but still under 8kB with a repeating soundtrack!

`File size of "dist/release.zip" is 8088/13312 bytes (60.75721153846154) %`

File: src/common/audio/themesong.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    shift(sht10, 0.3, +0, 0.1), // 2 - RHYTHM
+    shift(sht11, 0.3, +1, 0.1), // 3 - RHYTHM
+    shift(sht10, 0.3, +1, 0.1), // 4 - RHYTHM
+  ],
+  {
+    0: [0, 1, 0, 1],
+    1: [2, 3, 3, 4],
+    2: [4, 2, 1, 0],
+  }
+); // tslint:enable:no-magic-numbers
```
Comment (alexkindel) [inline src/common/audio/themesong.ts]: neat

File: src/common/audio/themesong.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+const ins0: GameSound = new GameSound('sawtooth', sht00.nreg());
+const ins1: GameSound = new GameSound('square', sht10.nreg());
+const ins2: GameSound = new GameSound('sine', sht10.nreg());
+
+// tslint:disable:no-magic-numbers
+export const gameMusic: SongHandler = new SongHandler(
+  [ins0, ins1, ins2],
+  [
+    // Sheet , volume set , octave shift , staccato set
```
Comment (alexkindel) [inline src/common/audio/themesong.ts]: Interesting stuff

File: src/common/audio/theory.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        ? NOTE_VALS[b]
+        : NOTE_VALS.C;
+    if (n.length > 1) {
+      if (n.substr(1, 1) === SHARP) {
+        m += 1;
+      } else if (n.substr(1, 1) === FLAT) {
+        m -= 1;
+      }
+    }
+    return Math.pow(2, (m - NOTE_VALS.A) / 12 + (o - 4)) * A440;
```
Comment (alexkindel) [inline src/common/audio/theory.ts]: tslint:enable:no-magic-mathematical-formulas

File: src/common/audio/soundeffects.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { state } from './../../index';
```
Comment (alexkindel) [inline src/common/audio/soundeffects.ts]: should be sound-effects.ts

File: src/common/audio/gamesound.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import { Sheet } from './music';
```
Comment (alexkindel) [inline src/common/audio/gamesound.ts]: should be game-sound.ts

File: src/common/audio/theory.ts (hunk 3 above)
Comment (QuantumBits) [reply src/common/audio/theory.ts]: but... all mathematical formulas are magic

**Pull request merged**